base/base.ts

```diff
@@ -80,6 +81,7 @@ export interface ComponentProps {
         | CustomOptionString;
     isSearchable?: boolean;
     format?: string;
+    whenChanged?: ((cbk: FrontendCBK, value?: any) => void | string);
```

Comment:
just curious why we name it `whenChanged?` instead of the regular `onChange` 😄  ?

Reply:
Haha good question. I found it keeps distinction in the client, because the ts interpreter was getting confused with component onChange methods. Additionally it makes it clear what is doing what, especically when you have an onChange method wrapping the whenChanged method.  e.g: 
`          onChange={(e, v) => {
            helpers.setValue(v);
            field.onChange(v);
            whenChanged?.()(formik);
          }}`